### ISSUE TITLE
checker panic recovery can mask incomplete results

### ISSUE THREAD

**Issue by luojiyin1987** [go-critic/go-critic#1533]
## Issue

In `cmd/go-critic/check.go:151-168` (and `cmd/gocritic/check.go`), the panic recovery mechanism has two problems:

### 1. Parent goroutine can exit before child re-panics

The current defer order is:
```go
defer func() {
    wg.Done()       // parent goroutine can proceed here
    <-sema
    r := recover()  // too late — parent may have called os.Exit
    panic(r)
}()
```

If `c.Check(f)` panics, `wg.Done()` is called before `recover()`. The parent goroutine can then proceed through `wg.Wait()` → output loop → `p.exit()` → `os.Exit`, masking the panic entirely. The lint result is incomplete with no error indication.

### 2. Unsafe type assertion in safeSizesInfoSizeof

`linter/linter.go:363` uses `r.(string)` without comma-ok guard. If the recovered value is not a string (e.g. `*runtime.TypeAssertionError`), this triggers a secondary panic that masks the original cause.

## Fix

1. Capture panics into a per-index slice, do cleanup and `wg.Done()` in child goroutine, re-throw after `wg.Wait()` in parent.
2. Change `r.(string)` to use comma-ok guard.
